### PR TITLE
west: openocd: find path intree

### PR DESCRIPTION
**Pull request by hakehuang**
when run package from another PC, the openocd path may not the same, so try to use ZEPHYR_BASE when
not aligned
and add LPC CMSIS-DAP support

### CODE REVIEW

**Comment by hakehuang** [comment]
@dleach02  can you check without this we can not support CI for lpcxpresso11u68.

**Comment by hakehuang** [comment]
@carlescufi  can you help to review

**Comment by dleach02** [comment]
> @dleach02 can you check without this we can not support CI for lpcxpresso11u68.

Looks fine but I'm not sure why 11u68 is an issue?

**Review by dleach02 — DISMISSED** [review]

**Comment by hakehuang** [comment]
> Looks fine but I'm not sure why 11u68 is an issue?

11u68 only supports openocd with NXP extension of it

File: scripts/pylib/twister/twisterlib/handlers.py [hunk 1]
```diff
@@ -512,6 +512,9 @@ def _create_command(self, runner, hardware):
                     elif runner == "openocd" and product == "EDBG CMSIS-DAP":
                         command_extra_args.append("--cmd-pre-init")
                         command_extra_args.append("cmsis_dap_serial %s" % board_id)
+                    elif runner == "openocd" and product == "LPC-LINK2 CMSIS-DAP":
+                        command_extra_args.append("--cmd-pre-init")
+                        command_extra_args.append("adapter serial %s" % board_id)
```
Comment (gchwier) [inline scripts/pylib/twister/twisterlib/handlers.py:517]: Could you please also apply this changes in:
https://github.com/zephyrproject-rtos/zephyr/blob/1cfe90d825a1e07d9aa77bf389338dc72dfda6fe/scripts/pylib/pytest-twister-harness/src/twister_harness/device/hardware_adapter.py#L64
there is same method but used by by pytest.

File: scripts/west_commands/runners/openocd.py [hunk 2]
```diff
@@ -40,7 +41,14 @@ def __init__(self, cfg, pre_init=None, reset_halt_cmd=DEFAULT_OPENOCD_RESET_HALT
                  target_handle=DEFAULT_OPENOCD_TARGET_HANDLE):
         super().__init__(cfg)
 
-        support = path.join(cfg.board_dir, 'support')
+        if not path.exists(cfg.board_dir):
+            # try to find the board support in-tree
+            cfg_board_path = path.normpath(cfg.board_dir)
+            _temp_path = cfg_board_path.split("boards/")[1]
+            support = path.join(environ['ZEPHYR_BASE'], "boards", _temp_path, 'support')
```
Comment (gchwier) [inline scripts/west_commands/runners/openocd.py:48]: think about using:
`from zephyr_ext_common import ZEPHYR_BASE`
like in intel_adsp.py.

File: scripts/pylib/twister/twisterlib/handlers.py (hunk 1 above)
Comment (hakehuang) [reply scripts/pylib/twister/twisterlib/handlers.py:517]: thanks to reminder this change, I add

**Comment by gchwier** [comment]
Twister part is ok for me and I can approve it.

But I have some objections about West part.
> when run package from another PC, the openocd path may not the same, so try to use ZEPHYR_BASE when
not aligned

you mean, when running Twister from another PC?
You're checking `cfg.board_dir` only for `openocd` board, but it can be more general problem, maybe better to change paths in Twister, before passing it to west, or in west, but when reading configs, to update this paths also for other boards.

Good to ask someone who is responsible for West to review it.
Nit: west commit is signed two times

**Comment by hakehuang** [comment]
> you mean, when running Twister from another PC?
> You're checking `cfg.board_dir` only for `openocd` board, but it can be more general problem, maybe better to change paths in Twister, before passing it to west, or in west, but when reading configs, to update this paths also for other boards.

only openocd need the this path for customized settings. I did not see other debugger has such requirements.

**Comment by hakehuang** [comment]
> Nit: west commit is signed two times

opps, let me fix it. Thanks for checking

**Review by dleach02 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**